app/models/ems_refresh/save_inventory_infra.rb

```diff
@@ -44,6 +44,19 @@ def save_ems_infra_inventory(ems, hashes, target = nil)
       return
     end
 
+    # Check if a vm was removed and we still have disk(s)
+    if target.kind_of?(VmOrTemplate) && hashes[:vms].blank?
+      current = hashes[:vm_disks].collect { |disk| URI(disk[:storage][:ems_ref]).path.split('/').last }.uniq
```

Comment:
vm_disks is already RHEV specific so when this key is not in the hashes (any other provider) it won't try to parse ems_ref.